src/app/OperationalSessionSetup.cpp

```diff
@@ -290,12 +291,14 @@ void OperationalSessionSetup::UpdateDeviceData(const Transport::PeerAddress & ad
     // Do not touch `this` instance anymore; it has been destroyed in DequeueConnectionCallbacks.
 }
 
-CHIP_ERROR OperationalSessionSetup::EstablishConnection(const ReliableMessageProtocolConfig & config)
+CHIP_ERROR OperationalSessionSetup::EstablishConnection(const ResolveResult & result)
 {
+    auto config = result.mrpRemoteConfig;
 #if INET_CONFIG_ENABLE_TCP_ENDPOINT
     // TODO: Combine LargePayload flag with DNS-SD advertisements from peer.
     // Issue #32348.
-    if (mTransportPayloadCapability == TransportPayloadCapability::kLargePayload)
+    if (mTransportPayloadCapability == TransportPayloadCapability::kLargePayload && result.supportsTcpServer &&
+        result.supportsTcpClient)
```

Comment:
Why do we care whether it supports TCP client?  We're going to be the client; as long as it supports server we are fine, no?

Reply:
good point, updated